runtime/compiler/runtime/J9Profiler.hpp

```diff
@@ -637,6 +637,8 @@ class TR_BlockFrequencyInfo
    bool    isJProfilingData() { return _counterDerivationInfo != NULL; }
    static int32_t *getEnableJProfilingRecompilation() { return &_enableJProfilingRecompilation; }
    static void    enableJProfilingRecompilation() { _enableJProfilingRecompilation = -1; }
+   void setIsQueuedForRecompilation() { _isQueuedForRecompilation = -1; }
```

Comment:
Why is this not a boolean?

Reply:
This flag is intended to use in JIT compiled code to check if method is Queued for recompilation. It will generate following trees. 
```
ificmpeq -> skipCode
      iload _isQueuedForRecompilation
      iconst -1
```
I am not sure a boolean here serves purpose. Do you have trees in mind which I can use here?

Reply:
I see. We should document this as it is non-obvious. Curious about the choice of `-1` though. I figure comparing against `0` would be more efficient on most platforms including Z where we can just test this value in memory without loading it in a register at all.

Reply:
On Z We can generate in memory compare immediate if we found the immediate value lies within the range in which we can encode it in the instruction so -1 would be in immediate field to compare with in memory. 
Verifying the instructions generated for above sequence on Z,
```
LARL R, &isQueuedForRecompilation
CHSI 0(R),0xFFFF
```
To answer your question, There is no particular reason behind using -1.

Reply:
Thanks for clarifying. I forgot about `CHSI`. Indeed it will work with most integers on Z.